Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM node:20-alpine
+
+COPY . /app
+WORKDIR /app
+
+RUN npm install -g pnpm
+RUN pnpm i
+RUN pnpm run build
+
+ENTRYPOINT ["sh", "-c", "pnpm next start"]
```

Comment:
No idea if there's a better way of building next.js docker apps - I tried the Dockerfile from the official guides, but they are using node 18, which is incompatible with this project, and there's a bunch of things I don't understand, so it naturally didn't work for me and I couldn't figure out how to fix it

This is the simplest working solution I could come up with

Reply:
can probably do pnpm start since its aliased to run next start in the `package.json`

Reply:
I also do CMD["thing here", "thing here"] on my production builds, not sure the difference between that end ENTRYPOINT. Normally in docker you'd also want to download the node version at the start, but since you're self-hosting the machine probably has node already and its somehow recognizing that

Reply:
another thing that doesn't really matter but if you _really_ want max efficiency is to use yarn to globally install pnpm so `RUN yarn global add pnpm && pnpm i`